ext/openssl/ossl_ssl.c

```diff
@@ -1406,6 +1406,24 @@ ossl_ssl_s_alloc(VALUE klass)
     return TypedData_Wrap_Struct(klass, &ossl_ssl_type, NULL);
 }
 
+static VALUE
+peer_ip_address(VALUE self)
+{
+    VALUE io = rb_funcall(self, rb_intern("to_io"), 0);
```

Comment:
Since we know `self`, we can access the IO object directly via `rb_attr_get(self, id_i_io)`.